tool/src/org/antlr/v4/tool/ErrorType.java

```diff
@@ -960,57 +960,21 @@
 	/**
 	 * Compiler Error 170.
 	 *
-	 * <pre>
-	 * mode M1;
-	 * A1: 'a'; // ok
-	 * mode M2;
-	 * A2: 'a'; // ok
-	 * M1: 'b'; // error 170
-	 * </pre>
+	 * <p>can not declare mode, token or channel with reserved name</p>
 	 *
-	 * <p>mode <em>name</em> conflicts with token with same name</p>
+	 * <p>Reserved names: HIDDEN, DEFAULT_TOKEN_CHANNEL, DEFAULT_MODE, SKIP, MORE, EOF, MAX_CHAR_VALUE, MIN_CHAR_VALUE.
 	 */
-	MODE_CONFLICTS_WITH_TOKEN(170, "mode <arg> conflicts with token with same name", ErrorSeverity.ERROR),
+	DECLARATION_CONFLICTS_WITH_COMMON_CONSTANTS(170, "cannot declare mode, token or channel with reserved name <arg>", ErrorSeverity.ERROR),
 	/**
 	 * Compiler Error 171.
 	 *
-	 * <p>can not use or declare token with reserved name</p>
-	 *
-	 * <p>Reserved names: HIDDEN, DEFAULT_TOKEN_CHANNEL, SKIP, MORE, MAX_CHAR_VALUE, MIN_CHAR_VALUE.
-	 *
-	 * <p>Can be used but cannot be declared: EOF</p>
-	 */
-	TOKEN_CONFLICTS_WITH_COMMON_CONSTANTS(171, "cannot use or declare token with reserved name <arg>", ErrorSeverity.ERROR),
-	/**
-	 * Compiler Error 172.
-	 *
-	 * <p>can not use or declare channel with reserved name</p>
-	 *
-	 * <p>Reserved names: DEFAULT_MODE, SKIP, MORE, EOF, MAX_CHAR_VALUE, MIN_CHAR_VALUE.
-	 *
-	 * <p>Can be used but cannot be declared: HIDDEN, DEFAULT_TOKEN_CHANNEL</p>
-	 */
-	CHANNEL_CONFLICTS_WITH_COMMON_CONSTANTS(172, "cannot use or declare channel with reserved name <arg>", ErrorSeverity.ERROR),
-	/**
-	 * Compiler Error 173.
-	 *
-	 * <p>can not use or declare mode with reserved name</p>
-	 *
-	 * <p>Reserved names: HIDDEN, DEFAULT_TOKEN_CHANNEL, SKIP, MORE, MAX_CHAR_VALUE, MIN_CHAR_VALUE.
-	 *
-	 * <p>Can be used and cannot declared: DEFAULT_MODE</p>
-	 */
-	MODE_CONFLICTS_WITH_COMMON_CONSTANTS(173, "cannot use or declare mode with reserved name <arg>", ErrorSeverity.ERROR),
-	/**
-	 * Compiler Error 174.
-	 *
-	 * <p>empty strings not allowed</p>
+	 * <p>string literals cannot be empty</p>
 	 *
 	 * <pre>A: '''test''';</pre>
 	 * <pre>B: '';</pre>
 	 * <pre>C: 'test' '';</pre>
 	 */
-	EMPTY_STRINGS_NOT_ALLOWED(174, "string literals cannot be empty", ErrorSeverity.ERROR),
+	EMPTY_STRINGS_NOT_ALLOWED(171, "string literals cannot be empty", ErrorSeverity.ERROR),
```

Comment:
❗️ Renumbering an existing error or warning is not an allowed change. After a warning or error is introduced, that number can never be used for any other purpose. See `ALL_OPS_NEED_SAME_ASSOC` and `LEXER_ACTION_PLACEMENT_ISSUE` for examples of old error numbers that are no longer reported by ANTLR 4.